plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/AppEngineStandardFacet.java

```diff
@@ -1,13 +1,246 @@
-  static final String INSTALL_ACTION_ID = "com.google.cloud.tools.eclipse.appengine.facet.install.action";
-  static final String UNINSTALL_ACTION_ID = "com.google.cloud.tools.eclipse.appengine.facet.uninstall.action";
+
+  /**
+   * Returns true if project has the App Engine Standard facet and false otherwise.
+   *
+   * @param project the project; project should not be null
+   * @return true if project has the App Engine Standard facet and false otherwise
+   */
+  public static boolean hasAppEngineFacet(IFacetedProject project) {
+    Preconditions.checkNotNull(project, "project is null");
+
+    IProjectFacet appEngineFacet = ProjectFacetsManager.getProjectFacet(ID);
+    return project.hasProjectFacet(appEngineFacet);
+  }
+
+  /**
+   * Returns true is <code>facetRuntime</code> is an App Engine Standard runtime and false otherwise
+   *
+   * @param facetRuntime the facet runtime; runtime should not be null
+   * @return true is <code>facetRuntime</code> is an App Engine Standard runtime and false otherwise
+   */
+  public static boolean isAppEngineRuntime(IRuntime facetRuntime) {
+    Preconditions.checkNotNull(facetRuntime, "runtime is null");
+
+    org.eclipse.wst.server.core.IRuntime serverRuntime = FacetUtil.getRuntime(facetRuntime);
+    if (serverRuntime != null) {
+      IRuntimeType runtimeType = serverRuntime.getRuntimeType();
+      return runtimeType.getId().equals(DEFAULT_RUNTIME_ID);
+    } else {
+      return false;
+    }
+  }
+
+  /**
+   * Returns true is <code>serverRuntime</code> is an App Engine Standard runtime and false otherwise
+   *
+   * @param serverRuntime the server runtime, runtime should not be null
+   * @return true if <code>serverRuntime</code> is an App Engine Standard runtime and false otherwise
+   */
+  public static boolean isAppEngineRuntime(org.eclipse.wst.server.core.IRuntime serverRuntime) {
+    Preconditions.checkNotNull(serverRuntime, "runtime is null");
+    IRuntimeType runtimeType = serverRuntime.getRuntimeType();
+    return runtimeType.getId().equals(DEFAULT_RUNTIME_ID);
+  }
+
+  /**
+   * Checks to see if <code>facetedProject</code> has the App Engine facet installed. If not, it installs
+   * the App Engine facet.
+   *
+   * @param facetedProject the workspace faceted project
+   * @param monitor the progress monitor
+   * @throws CoreException if anything goes wrong during install
+   */
+  public static void installAppEngineFacet(IFacetedProject facetedProject, boolean installDependentFacets, IProgressMonitor monitor)
+      throws CoreException {
+    // Install required App Engine facets i.e. Java 1.7 and Dynamic Web Module 2.5
+    if (installDependentFacets) {
+      installJavaFacet(facetedProject, monitor);
+      installWebFacet(facetedProject, monitor);
+    }
+
+    IProjectFacet appEngineFacet = ProjectFacetsManager.getProjectFacet(AppEngineStandardFacet.ID);
+    IProjectFacetVersion appEngineFacetVersion = appEngineFacet.getVersion(AppEngineStandardFacet.VERSION);
```

Comment:
Yes, the class name qualifies them. AppEngineStandardFacet.VERSION has everything we need